### PR TITLE
enhance dep_check advisor

### PR DESCRIPTION
**Pull request by NingW101**
- resolve related statements of required packages after removing find_package statement

### CODE REVIEW

File: src/advisors/dep_check.ts [hunk 1]
```diff
@@ -145,26 +150,63 @@ class DepCheckAdvisor implements IAdvisor {
...
+              if (pkgName !== "" && resolvedPkgs.includes(pkgName)) {
+                // comment this statement
+                actions.push(
+                  new FileChangeAction(
+                    proj.fileChangeManager,
+                    `Comment unnecessary statements for \`${pkgName}\` from \`CMakeLists.txt\` at line **${
+                      i + 1
+                    }**`,
+                    new FileRegion(file, i + 1),
+                    "# " + lines[i]
```
Comment (lingyuncai) [inline src/advisors/dep_check.ts:186]: If I have a below statement in cmake, according to the logic, line 3 & line 4 will be commented out. However, the end bracket `)` of `include_directories()` will also be commented out together, leading to syntax error. Shall we also consider such scenario together to make it right?

```cmake
1  include_directories(
2      ${CMAKE_SOURCE_DIR}/include
3      ${SDL2_INCLUDE_DIRS}
4      ${SDL2_TTF_INCLUDE_DIRS})
```

File: src/advisors/dep_check.ts [hunk 2]
```diff
@@ -145,26 +150,63 @@ class DepCheckAdvisor implements IAdvisor {
...
+                // emscripten and the compiler and linker flags for
+                // this package are added can be added into
+                // resolvedPkgs
+                resolvedPkgs.push(pkgName);
               } else {
                 nonPorts.push(m.groups.pkg);
               }
+            } else {
+              const n = lines[i].match(libraryReg);
+              const o = lines[i].match(includeDirReg);
```
Comment (lingyuncai) [inline src/advisors/dep_check.ts:163]: We may want to avoid naming variables to `n` and `o` as much as possible.

**Review by lingyuncai — COMMENTED** [review]
Thanks Ning, some comments to be addressed :)

File: src/advisors/dep_check.ts (hunk 1 above)
Comment (NingW101) [reply src/advisors/dep_check.ts:186]: Exactly, working on it

File: src/advisors/dep_check.ts (hunk 2 above)
Comment (NingW101) [reply src/advisors/dep_check.ts:163]: Got it, working on it

File: src/advisors/dep_check.ts [hunk 3]
```diff
@@ -145,26 +153,90 @@ class DepCheckAdvisor implements IAdvisor {
...
+                  });
+                  if (isFatal) {
+                    actions.push(
+                      new FileChangeAction(
+                        proj.fileChangeManager,
+                        `Comment useless statement for \`${libName}\` from \`CMakeLists.txt\` at line **${
+                          i + 1
+                        }**`,
+                        new FileRegion(file, i + 1),
+                        `#` + lines[i]
```
Comment (lingyuncai) [inline src/advisors/dep_check.ts:191]:
```suggestion
                        `# ${lines[i]}`
```

File: src/advisors/dep_check.ts [hunk 4]
```diff
@@ -145,26 +153,90 @@ class DepCheckAdvisor implements IAdvisor {
...
-            );
+        }
+
+        if (uselessParamsLocationMap.size > 0) {
+          const tableHeader = `| Line | Parameter |\n|:---:|:---:|`;
+          const tableContent = Array.from(uselessParamsLocationMap.entries())
+            .map(([line, param]) => `| _**${line}**_ | \`${param}\``)
+            .join("\n");
+          const descPrefix = `Followings are the parameters and their location in \`CMakeLists.txt\` file,  which will be invalid after you click apply recipe button because webinizer will attempt to remove \`find_package\` for this package since it has been ported by emscripten.`;
+          const descSuffix = `You could resolve them manually to avoid the possible interruption of cmake construction.`;
```
Comment (lingyuncai) [inline src/advisors/dep_check.ts:231]:
```suggestion
          const descSuffix = `You could resolve them manually to avoid possible build failures.`;
```

**Review by lingyuncai — APPROVED** [review]
Thanks Ning! LGTM~

**Pull request merged**